Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -578,72 +560,55 @@ public enum DefaultFunctionOpacityOptions {
+        { WarnShadowing, OptionCompatibility.OptionLibraryImpliesLocalWarning },
+        { UseStandardLibraries, OptionCompatibility.OptionLibraryImpliesLocalError },
       }
     );
-    DooFile.RegisterNoChecksNeeded(
-      WarnAsErrors,
-      ProgressOption,
-      LogLocation,
-      LogLevelOption,
-      ManualTriggerOption,
-      ShowHints,
-      Check,
-      Libraries,
-      Output,
-      PluginOption,
-      Prelude,
-      Target,
-      Verbose,
-      AllowDeprecation,
-      FormatPrint,
-      JsonDiagnostics,
-      QuantifierSyntax,
-      SpillTranslation,
-      StdIn,
-      TestAssumptions,
-      WarnShadowing,
-      ManualLemmaInduction,
-      TypeInferenceDebug,
-      GeneralTraits,
-      GeneralNewtypes,
-      TypeSystemRefresh,
-      VerificationLogFormat,
-      VerifyIncludedFiles,
-      DisableNonLinearArithmetic,
-      NewTypeInferenceDebug,
-      UseBaseFileName,
-      WarnMissingConstructorParenthesis,
-      UseJavadocLikeDocstringRewriterOption,
-      IncludeRuntimeOption,
-      InternalIncludeRuntimeOptionForExecution,
-      WarnContradictoryAssumptions,
-      WarnRedundantAssumptions,
-      VerificationCoverageReport,
-      NoTimeStampForCoverageReport,
-      DefaultFunctionOpacity,
-      UseStandardLibraries,
-      OptimizeErasableDatatypeWrapper,
-      AddCompileSuffix,
-      SystemModule,
-      ExecutionCoverageReport,
-      ExtractCounterexample,
-      ShowProofObligationExpressions
-      );
+    DooFile.RegisterNoChecksNeeded(WarnAsErrors, false);
+    DooFile.RegisterNoChecksNeeded(ProgressOption, false);
+    DooFile.RegisterNoChecksNeeded(LogLocation, false);
+    DooFile.RegisterNoChecksNeeded(LogLevelOption, false);
+    DooFile.RegisterNoChecksNeeded(ManualTriggerOption, true);
+    DooFile.RegisterNoChecksNeeded(ShowHints, false);
+    DooFile.RegisterNoChecksNeeded(Libraries, true);
+    DooFile.RegisterNoChecksNeeded(Output, false);
+    DooFile.RegisterNoChecksNeeded(PluginOption, false);
+    DooFile.RegisterNoChecksNeeded(Prelude, false);
+    DooFile.RegisterNoChecksNeeded(Target, false);
+    DooFile.RegisterNoChecksNeeded(Verbose, false);
+    DooFile.RegisterNoChecksNeeded(JsonDiagnostics, false);
+    DooFile.RegisterNoChecksNeeded(QuantifierSyntax, true);
+    DooFile.RegisterNoChecksNeeded(SpillTranslation, false);
+    DooFile.RegisterNoChecksNeeded(StdIn, false);
+    DooFile.RegisterNoChecksNeeded(TestAssumptions, false);
+    DooFile.RegisterNoChecksNeeded(ManualLemmaInduction, true);
+    DooFile.RegisterNoChecksNeeded(TypeInferenceDebug, false);
+    DooFile.RegisterNoChecksNeeded(GeneralTraits, false);
+    DooFile.RegisterNoChecksNeeded(GeneralNewtypes, false);
+    DooFile.RegisterNoChecksNeeded(TypeSystemRefresh, false);
+    DooFile.RegisterNoChecksNeeded(VerificationLogFormat, false);
+    DooFile.RegisterNoChecksNeeded(VerifyIncludedFiles, false);
+    DooFile.RegisterNoChecksNeeded(DisableNonLinearArithmetic, true);
```

Comment:
Some of these are tough to be super confident on, but given being overly conservative and recording anything that MIGHT be semantic is pretty much harmless, defaulting to `true` in those cases is pretty safe (and that might even be worth adding in comments to the `semantic` flag)

Reply:
It's not harmless to have `true` when it should be `false`. If you set `TypeSystemRefresh` to true then you're going to have a bad time because part of the code could resolve with the new and part with the old resolver.